bsp/src/mill/bsp/BSP.scala

```diff
@@ -73,15 +58,16 @@ object BSP extends ExternalModule {
   def createBspConnectionJson(): String = {
     val millPath = sys.props
       .get("java.class.path")
-      .getOrElse(throw new IllegalStateException("System property java.class.path not set"))
+      .getOrElse(throw new IllegalStateException(
+        "System property java.class.path not set"))
 
     write(
       BspConfigJson(
         "mill-bsp",
         Seq(
           "sh",
           "-c",
-          s"env ${sys.env.map { case (k, v) => s""""$k=$v"""" }.toSeq.mkString(" ")} $millPath -i mill.contrib.BSP/start"
+          s"env ${sys.env.map { case (k, v) => s""""$k=$v"""" }.toSeq.mkString(" ")} $millPath -i ${BSP.getClass.getCanonicalName.split("\\$").last}/start"
```

Comment:
@joan38  So I've started to play around a bit with this in Metals, and one thing that surprised me a bit was that this captures all env variables and adds them to the `mill.json`. My worry is, especially in Metals if we automatically generation this for a mill build is that a user may accidentally commit this. I've fairly positive this will happen. I already see it happening a bunch with the new `.bsp/sbt.json` files that sbt has been producing. Unknowingly I expect users will be committing this with env variables such as passwords etc that they may obviously want.

In my experience with bsp discovery and the other build servers, I haven't seen this before. Is this a limitation of Mill, is this really necessary to include to start the server?

Reply:
Why are these env varibables explicitly set? Is it solving some known issue, e.g. a mill specific env varibale? In that case, we might better special handle a know case than bulk materializing a given state of env variables into this file.

Reply:
We initially copied all the env vars to pick stuff such as `COURSIER_REPOSITORY` or `JAVA_OPTS` and make sure the BSP compilation is equal to a cmd compilation.
That can be reverted easily.

Reply:
Ditto https://github.com/lihaoyi/mill/pull/984